src/backend/optimizer/plan/planner.c

```diff
@@ -473,8 +473,9 @@ standard_planner(Query *parse, const char *query_string, int cursorOptions,
 
 	/*
 	 * GPDB: allow to use parallel or not.
+	 * SINGLENODE_FIXME: We'll enable parallel in singlenode mode later.
 	 */
-	if (!enable_parallel)
+	if (!enable_parallel || GP_ROLE_SINGLENODE == Gp_role)
```

Comment:
If we login with utility mode, parallel is enable here. Maybe we should disable utility mode on singlenode.

Reply:
Actually, I have the same feeling with you about utility mode. PG don't have utility mode, why should we support it in GP singlenode mode.
Simply by check role in `assign_gp_role` can fix it. I just tried.

But it will cause so many tests change especially in isolation2. Grammar like `1U:` allow isolation tests connect to gpdb in utility mode. If we disable it, all the tests will be broken.

So my final decision in this pr is allowing utility mode temporally. What do you think?

Reply:
And about this issue, after looking around the code, I found currently parallel is actually not gonna happened in utility mode.  useMppParallelMode will only be set to true if there exists motion.

But for this line, I think we can keep it. To clearly show that we don't support parallel in SINGLNODE mode now, but should support it in the future.

Reply:
ok，we can open an issue and solve this later.

Reply:
https://github.com/cloudberrydb/cloudberrydb/issues/123